packages/sanity/src/desk/panes/document/documentPanel/DeletedDocumentBanner.tsx

```diff
@@ -1,46 +1,49 @@
-import React, {useCallback, useMemo} from 'react'
+import React, {useCallback} from 'react'
 import {Button, Card, Container, Flex, Text} from '@sanity/ui'
 import {ReadOnlyIcon} from '@sanity/icons'
 import styled from 'styled-components'
 import {useDocumentPane} from '../useDocumentPane'
-import {useDocumentOperation, useTimelineSelector} from 'sanity'
+import {useDocumentOperation} from 'sanity'
+import {useRouter} from 'sanity/router'
 
 const Root = styled(Card)`
   position: relative;
   z-index: 50;
 `
-export function DeletedDocumentBanner() {
-  const {documentId, documentType, timelineStore} = useDocumentPane()
-  const chunks = useTimelineSelector(timelineStore, (state) => state.chunks)
-  const lastRevisionId = useMemo(() => {
-    const lastRevision = chunks.filter((chunk) => !['delete', 'initial'].includes(chunk.type))
 
-    return lastRevision?.[0]?.id
-  }, [chunks])
+interface DeletedDocumentBannerProps {
+  revisionId?: string | null
+}
+
+export function DeletedDocumentBanner({revisionId}: DeletedDocumentBannerProps) {
+  const {documentId, documentType} = useDocumentPane()
   const {restore} = useDocumentOperation(documentId, documentType)
+  const {navigateIntent} = useRouter()
   const handleRestore = useCallback(() => {
-    restore.execute(lastRevisionId)
-  }, [restore, lastRevisionId])
+    if (revisionId) {
+      restore.execute(revisionId)
+      navigateIntent('edit', {id: documentId, type: documentType})
+    }
+  }, [documentId, documentType, navigateIntent, restore, revisionId])
 
   return (
     <Root data-testid="deleted-document-banner" shadow={1} tone="transparent">
-      <Container paddingX={4} paddingY={lastRevisionId ? 2 : 3} sizing="border" width={1}>
+      <Container paddingX={4} paddingY={revisionId ? 2 : 3} sizing="border" width={1}>
         <Flex align="center">
           <Text size={1}>
             <ReadOnlyIcon />
           </Text>
 
           <Flex align="center" gap={2} flex={1} marginLeft={3}>
-            <Text size={1}>This document is deleted and can’t be edited.</Text>
-            {lastRevisionId && (
+            <Text size={1}>This document has been deleted.</Text>
```

Comment:
**suggestion**: use terser language, it should (hopefully) be self evident that the document can't be edited!